core/block_validator.go

```diff
@@ -69,6 +69,11 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 		return fmt.Errorf("transaction root hash mismatch (header value %x, calculated %x)", header.TxHash, hash)
 	}
 
+	// Verify aggregated signature for BLS transactions in a block, if any
+	if err := v.engine.VerifyAggregate(block); err != nil {
```

Comment:
I'm not sure if `VerifyAggregate` should go in `ValidateBody` or `FinalizeAndAssemble`.

By the function description of `ValidateBody` it seems more appropriate since the `Body` of a `Block` contains `Transactions`, which is the data that we are working with.

Whereas, `FinalizeAndAssemble` seems to do post checks more broadly